compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -337,47 +337,26 @@ impl<'hir> LoweringContext<'_, 'hir> {
 
     fn lower_expr_if(
         &mut self,
-        span: Span,
         cond: &Expr,
         then: &Block,
         else_opt: Option<&Expr>,
     ) -> hir::ExprKind<'hir> {
         // FIXME(#53667): handle lowering of && and parens.
 
-        // `_ => else_block` where `else_block` is `{}` if there's `None`:
-        let else_pat = self.pat_wild(span);
-        let (else_expr, contains_else_clause) = match else_opt {
-            None => (self.expr_block_empty(span), false),
-            Some(els) => (self.lower_expr(els), true),
-        };
-        let else_arm = self.arm(else_pat, else_expr);
+        let then_blk = self.lower_block(then, false);
```

Comment:
I guess the current error:

```
error: `let` expressions are not supported here
   --> library/core/src/time.rs:567:12
    |
567 |         if let Some(s) = self.secs.checked_mul(rhs as u64) {
```

Happens because `if let` expressions are not being lowered in this method.

Reply:
`if let` expressions should still be lowered to `match` for now, as the current code does.

Reply:
Thanks! Now `./x.py test --bless` is working :D